Revert of usage of newer CoreAudio functions inspired by VLC commit:

http://mailman.videolan.org/pipermail/vlc-devel/2008-April/042612.html

--- libao2/ao_coreaudio.c
+++ libao2/ao_coreaudio.c
@@ -75,8 +75,6 @@
   int b_digital;                            /* Are we running in digital mode? */
   int b_muted;                              /* Are we muted in digital mode? */
 
-  AudioDeviceIOProcID renderCallback;       /* Render callback used for SPDIF */
-
   /* AudioUnit */
   AudioUnit theOutputUnit;
 
@@ -814,12 +812,11 @@
 
 
     /* Create IOProc callback. */
-    err = AudioDeviceCreateIOProcID(ao->i_selected_dev,
+    err = AudioDeviceAddIOProc(ao->i_selected_dev,
                                     (AudioDeviceIOProc)RenderCallbackSPDIF,
-                                    (void *)ao,
-                                    &ao->renderCallback);
+                                    (void *)ao);
 
-    if (err != noErr || ao->renderCallback == NULL)
+    if (err != noErr)
     {
         ao_msg(MSGT_AO, MSGL_WARN, "AudioDeviceAddIOProc failed: [%4.4s]\n", (char *)&err);
         goto err_out1;
@@ -1113,12 +1110,12 @@
   }
   else {
       /* Stop device. */
-      err = AudioDeviceStop(ao->i_selected_dev, ao->renderCallback);
+      err = AudioDeviceStop(ao->i_selected_dev, (AudioDeviceIOProc)RenderCallbackSPDIF);
       if (err != noErr)
           ao_msg(MSGT_AO, MSGL_WARN, "AudioDeviceStop failed: [%4.4s]\n", (char *)&err);
 
       /* Remove IOProc callback. */
-      err = AudioDeviceDestroyIOProcID(ao->i_selected_dev, ao->renderCallback);
+      err = AudioDeviceRemoveIOProc(ao->i_selected_dev, (AudioDeviceIOProc)RenderCallbackSPDIF);
       if (err != noErr)
           ao_msg(MSGT_AO, MSGL_WARN, "AudioDeviceRemoveIOProc failed: [%4.4s]\n", (char *)&err);
 
@@ -1176,7 +1173,7 @@
     }
     else
     {
-        err = AudioDeviceStop(ao->i_selected_dev, ao->renderCallback);
+        err = AudioDeviceStop(ao->i_selected_dev, (AudioDeviceIOProc)RenderCallbackSPDIF);
         if (err != noErr)
             ao_msg(MSGT_AO, MSGL_WARN, "AudioDeviceStop failed: [%4.4s]\n", (char *)&err);
     }
@@ -1201,7 +1198,7 @@
     }
     else
     {
-        err = AudioDeviceStart(ao->i_selected_dev, ao->renderCallback);
+        err = AudioDeviceStart(ao->i_selected_dev, (AudioDeviceIOProc)RenderCallbackSPDIF);
         if (err != noErr)
             ao_msg(MSGT_AO, MSGL_WARN, "AudioDeviceStart failed: [%4.4s]\n", (char *)&err);
     }
